modules/modes/puzzle_solver.py

```diff
@@ -415,6 +433,79 @@ def path():
                         yield from walk_one_tile("Down")
                         yield from walk_one_tile("Up")
 
+            # Glass Workshop
+            case MapRSE.ROUTE113_GLASS_WORKSHOP:
+                context.message = "Collecting ashes..."
+                use_repel = True
+                assert_item_exists_in_bag(
+                    "Soot Sack",
+                    "This mode requires the Soot Sack to have been obtained.",
+                )
+
```

Comment:
Maybe we can check here if we already have the flute in our inventory too

Reply:
Actually forgot about checking the is the flute is already in your inventory. Because you can if you want farm flutes. So we need to give the opportunity for the player to have 50 flutes.

I think it would be a quick win to hardcode the ash reward table, and farm the number of ashes depending on the selected item. We can do it with an input like we already have in the project like this one.

![image](https://github.com/user-attachments/assets/6f50bf50-f597-4f48-b918-20e9a1cc7a3e)

Since the table loot is not very big it won't take the whole screen

![image](https://github.com/user-attachments/assets/be5e75c8-5886-4ba8-967a-475285ce5c6a)

Another thing would be the automation of actually buying the desired item, but I think it can be done later